### PR TITLE
Add basic export function app

### PR DESCRIPTION
**Pull request by graham-skylight**
Adds a basic function app to wrap the FHIR API Server $export endpoint

### CODE REVIEW

File: src/FunctionApps/FHIRExport/Export/export_test.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    mock_get.assert_called_with(
+        "some-endpoint", headers={"Authorization": "Bearer some-token"}
+    )
+
+
+@mock.patch("requests.get")
+def test_poll_failure(mock_get):
+    """Status endpoint fails the first time"""
+    mock_get.return_value = mock.Mock(ok=False, status_code=500, text="kablamo")
+    with pytest.raises(Exception):
```
Comment (bryan-skylight) [inline src/FunctionApps/FHIRExport/Export/export_test.py:65]: Can you add a comment about how this test works and how it ensures the failures are handled properly?

File: src/FunctionApps/FHIRExport/Export/export_test.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    assert "{}" == resp.text
+
+    mock_sleep.assert_called_with(POLLING_FREQUENCY)
+
+
+@mock.patch("requests.get")
+@mock.patch("time.sleep")
+def test_poll_timeout(mock_sleep, mock_get):
+    mock_get.return_value = mock.Mock(status_code=202)
+    with pytest.raises(Exception):
```
Comment (bryan-skylight) [inline src/FunctionApps/FHIRExport/Export/export_test.py:90]: Can you add a comment about how this test works and how it ensures the failures are handled properly?

File: src/FunctionApps/FHIRExport/Export/__init__.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+import os
+import time
+
+import requests
+import azure.functions as func
+
+
+# The time between polling requests in seconds
+POLLING_FREQUENCY = 2.5
+POLLING_RETRIES = 120  # 500ms * 120 retries == 5 min
```
[the remark this answers is not in the document]
Comment (graham-skylight) [reply src/FunctionApps/FHIRExport/Export/__init__.py:11]: woops - I had it at 500ms and then realized that azure probably can't get this done in a minute

File: src/FunctionApps/FHIRExport/Export/export_test.py (hunk 1 above)
Comment (graham-skylight) [reply src/FunctionApps/FHIRExport/Export/export_test.py:69]: Added a basic rationale - let me know if that helps out?

**Review by bryan-skylight — APPROVED** [review]

**Comment by bmaden-usds** [comment]
So far, the test workflow doesn't auto-detect new function apps.  To get the tests to run, please add the function app directory name `FHIRExport` to the list of names at around line 22 of `.github/workflows/tests.yaml`.

**Review by JSkeets — APPROVED** [review]
LGTM

**Pull request merged**